### DISCUSSION TITLE
How we run this

### DISCUSSION THREAD

**Discussion by zeuchste** [nebulastream/nebulastream#29] [Ideas]
As we have not discussed the details of our collaboration, I am starting this discussion, which should describe our procedures such that we can conclude a general frame for our collaboration.

test

**Comment by zeuchste** [comment]
### General Agreements

- We agree that the person who starts a discussion is responsible for finishing it and driving it
- The owner of a discussion also sets the deadlines and summarizes the results
- Even if meetings are done in a meeting slot, the result must have to be summarized in the discussion for the not-attending people
- We use gitbub discussions to sync on stuff not the slack channel
- We agree on three different entities to structure our working
  - EPICs represent one larger topic and is a one to one mapping to a gitbub discussion
  - Sub-EPIC represents one comment-thread within an EPIC
  - Issues are grouped under Sub-EPICs and reflect an item that is possible to finish within one sprint

**Reply by zeuchste** [reply]
also https://github.com/nebulastream/nebulastream-public/discussions/6#discussioncomment-9749582 is related here but I included them

**Reply by keyseven123** [reply]
To your second point: Do you mean hard deadlines? Or soft ones, e.g. if the discussion reignites because someone new enters it.

**Reply by zeuchste** [reply]
It is up to discussion, I mean rather he decides if the discussion can be closed as we reached some common sense and everybody had the chance to contribute (e.g. we will close in 2 days if nobody comments anymore)

**Reply by alepping** [reply]
Concerning 'We agree on three different entities to structure our working'
What about design documents?
A general flow would be nice (potentially mermaid flow chart). When to create a discussion, when not to, when to create a design document, when to create EPICs and issues from epics (already discussed here), etc.

**Reply by zeuchste** [reply]
yes we should do this

**Reply by zeuchste** [reply]
I can create this
- design document when?, 
- discussion RFC? Head of discussion based from discussion
- Discussions morph into design documents and be moved to codebase markdown file
-- what are the structural elements?

**Comment by zeuchste** [comment]
### From discussion to issue

For every discussion, we create one EPIC, then we create for each comment thread, once the discussion is over, a sub-epic, and then filled with one to multiple issues.

**Reply by adrianmichalke** [reply]
Sounds ok to me, but Im not sure if we really find a general rule here. It would be good if (sub-)epics are in some sense of equal size. In contrast, discussions and their comments have quite different scales and outcomes (e.g. some also lead to nothing).

**Reply by zeuchste** [reply]
I mean this is rather a proposal that of course can be different in certain situations but is rather the default, maybe some sub-discussions will lead nowhere and thus will never make it into a sub-epic.

**Reply by zeuchste** [reply]
I would in addition propose that we a) hide a sub-discussion if it is closed and end it with the sentence

"/closed, covered in issue #123"

**Comment by zeuchste** [comment]
### Milestone Meeting

Here, we have a degree of freedom what we expect before the milestone meeting:
- Solution 1: All issues for a milestone have to be specified before the milestone starts
- Solution 2: All sub-epics but not their issues have to be specified before the milestone starts

In addition, as we cannot foresee all details, issues can be created within a milestone

**Reply by keyseven123** [reply]
Maybe you could create a poll if there are two options. Then we can vote for one.

**Reply by zeuchste** [reply]
how do I do a poll here :)?

**Reply by keyseven123** [reply]
I think that you can not do it directly here. But you could create a poll in the general discussions.

**Reply by zeuchste** [reply]
Discussion result: We decided that once we agreed on the sub-epics, we will determine the deadline of the milestone

**Comment by zeuchste** [comment]
### Sprint

We agree of a sprint of two weeks for now, all issues for a sprint have to be ready at the sprint begin. The question is if and how we groom them

**Reply by keyseven123** [reply]
What do you mean by groom them? Discuss them in detail assign them to people?

**Reply by zeuchste** [reply]
We can discuss this, for me grooming is discussing the issue if there are open questions. Scheduling is then done afterwards

**Reply by keyseven123** [reply]
Sure, I am fine with iterating over the issues once at the beginning of a sprint. I would not expect there to be many open questions, as we have already discussed them in the github discussions.

**Reply by zeuchste** [reply]
Out of discussion: Create a manual for how to run the sprint @kisa7

**Comment by zeuchste** [comment]
### Component owners for the first milestone
(Taken from the slide that we put together)

- Testing: @adrianmichalke @klauck 
- Node Engine: @zeuchste @keyseven123 @ls-1801 
- Worker: @kisa7 @zeuchste @keyseven123 @ls-1801 
- Source/Sink: @alepping @klauck 
- Configuration: @ls-1801 @kisa7 
- Operators/Compiler: @alepping @adrianmichalke 
- DevOps: @keyseven123 @ls-1801 @klauck 
- User Stories: @kisa7 
- Plugin System: @alepping @zeuchste

**Comment by zeuchste** [comment]
### Everyday structure

I suggest a stand-up every day at 10 am to kickstart the day and keep everybody in snyc

**Reply by keyseven123** [reply]
I like this idea with the addition of it being 5-10 minutes and we discuss what everyone did yesterday and what everyone will do today.

**Reply by klauck** [reply]
plus a discussion of blockers (https://www.youtube.com/watch?v=er9gntPjTJU)

**Reply by keyseven123** [reply]
What do you mean by a discussion of blockers?

**Reply by zeuchste** [reply]
I think you need to watch the video for this answer :)

**Reply by keyseven123** [reply]
@klauck cool playlist you discovered there. I found it quite interesting and a good refresher.
